### PR TITLE
Update roadmap

### PR DESCRIPTION
**Pull request by phanimarupaka**

### CODE REVIEW

File: docs/ROADMAP.md [hunk 1]
```diff
@@ -4,15 +4,15 @@ Last updated: September 14th, 2021
```
Comment (mikebz) [inline docs/ROADMAP.md:4]: last updated needs to be updated

File: docs/ROADMAP.md [hunk 2]
```diff
@@ -4,15 +4,15 @@ Last updated: September 14th, 2021
 
 Draft of *v1* is released. Please follow the [installation](https://kpt.dev/installation/) guide.
 
-### 1. Declarative function pipeline
+### Declarative function pipeline
```
Comment (mikebz) [inline docs/ROADMAP.md:7]: Let's move all the delivered things in order (by when they were delivered) at the bottom.  

The sections should be
1) what's coming up next in chronological order
2) Historical roadmap

File: docs/ROADMAP.md (hunk 1 above)
Comment (phanimarupaka) [reply docs/ROADMAP.md:4]: Updated. Thanks.

File: docs/ROADMAP.md (hunk 2 above)
Comment (phanimarupaka) [reply docs/ROADMAP.md:7]: Updated.

**Comment by mikebz** [comment]
Please do this structure:

1) upcoming features by date, soonest release at the top and then going down from there.
Title: Upcoming features
Every theme and feature should have a bold headline and then a paragraph describing what is there.  If there are links to issues please include them at the bottom.

2) section at the bottom that just points to releases so that people can examine those by themselves, no need to copy those over.

v1 deliverables should be removed.  They are old news.

**Comment by mikebz** [comment]
Estimated completion date formatting should be consistent among all the items.

**Comment by mikebz** [comment]
I'd rename the 
Released features
to 
Latest release

**Comment by mikebz** [comment]
Release notes for things beyond the latest release can then go right after that section.  So we have:

1. Latest release
2. Release notes -> Prior releases
3. Upcoming features

Upgrading from previous version of kpt. can go away.

**Comment by phanimarupaka** [comment]
@mikebz Done.

**Review by mikebz — APPROVED** [review]
LGTM

**Pull request merged**